src/Bridge/Doctrine/Orm/SubresourceDataProvider.php

```diff
@@ -79,82 +80,10 @@ public function getSubresource(string $resourceClass, array $identifiers, array
-
-            $manager = $this->managerRegistry->getManagerForClass($identifierResourceClass);
-
-            if (!$manager instanceof EntityManagerInterface) {
-                throw new RuntimeException("The manager for $identifierResourceClass must be an EntityManager.");
-            }
-
-            $classMetadata = $manager->getClassMetadata($identifierResourceClass);
-
-            if (!$classMetadata instanceof ClassMetadataInfo) {
-                throw new RuntimeException("The class metadata for $identifierResourceClass must be an instance of ClassMetadataInfo.");
-            }
-
-            $qb = $manager->createQueryBuilder();
-            $alias = $queryNameGenerator->generateJoinAlias($identifier);
-            $relationType = $classMetadata->getAssociationMapping($previousAssociationProperty)['type'];
-            $normalizedIdentifiers = isset($identifiers[$identifier]) ? $this->normalizeIdentifiers($identifiers[$identifier], $manager, $identifierResourceClass) : [];
-
-            switch ($relationType) {
-                //MANY_TO_MANY relations need an explicit join so that the identifier part can be retrieved
-                case ClassMetadataInfo::MANY_TO_MANY:
-                    $joinAlias = $queryNameGenerator->generateJoinAlias($previousAssociationProperty);
-
-                    $qb->select($joinAlias)
-                        ->from($identifierResourceClass, $alias)
-                        ->innerJoin("$alias.$previousAssociationProperty", $joinAlias);
-
-                    break;
-                case ClassMetadataInfo::ONE_TO_MANY:
-                    $mappedBy = $classMetadata->getAssociationMapping($previousAssociationProperty)['mappedBy'];
-
-                    // first pass, o.property instead of alias.property
-                    if (null === $previousQueryBuilder) {
-                        $originAlias = "$originAlias.$mappedBy";
-                    } else {
-                        $previousAlias = "$previousAlias.$mappedBy";
-                    }
-
-                    $qb->select($alias)
-                        ->from($identifierResourceClass, $alias);
-                    break;
-                default:
-                    $qb->select("IDENTITY($alias.$previousAssociationProperty)")
-                        ->from($identifierResourceClass, $alias);
-            }
-
-            // Add where clause for identifiers
-            foreach ($normalizedIdentifiers as $key => $value) {
-                $placeholder = $queryNameGenerator->generateParameterName($key);
-                $qb->andWhere("$alias.$key = :$placeholder");
-                $queryBuilder->setParameter($placeholder, $value);
-            }
-
-            // recurse queries
-            if (null === $previousQueryBuilder) {
-                $previousQueryBuilder = $qb;
-            } else {
-                $previousQueryBuilder->andWhere($qb->expr()->in($previousAlias, $qb->getDQL()));
-            }
```

Comment:
this is recursive already no? Not sure what was going on wrong here 😸

Reply:
No, this was not recursive (despite the comment!).

Reply:
Should've been ? 😄

Reply:
Yup! :D